### DISCUSSION TITLE
Need help

### DISCUSSION THREAD

**Discussion by subham8907** [uBlockOrigin/uBOL-home#195] [Q&A] [answered]
Dear gorhill,

I hope this message finds you well. I'm reaching out because I'm developing a browser extension to remove tracking parameters, and I've run into some challenges that I hope you might be able to provide insight on.

My extension currently uses a JSON file containing approximately 800 static rules for identifying and removing tracking parameters. However, I'm encountering a peculiar issue with these rules. When all 800 rules are in a single ruleset, only a few of them seem to be working. Interestingly, if I split these rules into different rulesets, all of them work correctly. 

I'm aware that there's a limit to how many rulesets can be added in the manifest file, which presents a challenge for this workaround. I've noticed that in your implementation of uBlock Origin Lite, you're able to add around 500 rules in a single JSON file, and they all appear to be working correctly.

Given this context, I have some specific questions:

1. Do you have any insights into why rules might not work when combined in a large set, but function correctly when split up?
2. How do you manage to have all rules working in uBlock Origin Lite with a large ruleset in a single file?
3. Are there any specific techniques or data structures you use in uBlock Origin Lite that help with processing a large number of rules efficiently?
4. Do you have any recommendations for managing and processing a large number of rules without splitting them into multiple rulesets?

I've been looking at uBlock Origin Lite as a potential model for a more efficient approach, particularly in how you handle large rulesets and maintain performance.

. I'm committed to maintaining the open-source nature of the project and adhering to the license requirements.

Any advice or insights you could offer would be incredibly helpful. I have great respect for your work on uBlock Origin, and I would be very grateful for any guidance you could provide.

Thank you for your time and consideration.

Best regards,
Subham

**Comment by gorhill** [comment]
Maybe you are affected by <https://github.com/w3c/webextensions/issues/468>. Otherwise, best that you report to browser vendors with all the details on how they can reproduce your issue.

**Reply by subham8907** [reply, marked as answer]
If i share my code in github please guide me how to do it maybe you are able to figure it out i also want to make opensource anyway

**Reply by subham8907** [reply]
Please please

**Comment by subham8907** [comment]
If i share code in github can you look into it please it is important i mean it
